### ISSUE TITLE
Phantom /s messages

### ISSUE THREAD

**Issue by wight-airmash** [wight-airmash/ab-server#51]
From time to time, I notice /s messages from nowhere in different parts of the map. More often in the spectate mode. However, there are no prowlers nearby that could cause such an effect.

**Comment by congratulatio** [comment]
The culprit may be the recent frontend switch to [airmash-refugees/airmash-frontend](https://github.com/airmash-refugees/airmash-frontend), it's possible a bug was introduced while refactoring to a more readable codebase. 

The previous frontend, which has only minimal changes to the original minified engine.js, is still available at https://old.airmash.online/ for testing.

**Comment by wight-airmash** [comment]
It's also actual for starmash, even old starmash code (I use not the last version, without snow).

**Comment by derpsderps69** [comment]
Just chirping in.. this is def older than the frontend switch, although we could easily patch over it from there.

Wight mentioned doing a nicer protocol debugger for the frontend, I think that could help a lot if people could just hit "@" or something, and window.open() a log of last N messages or somesuch

**Comment by derpsderps69** [comment]
From frontend Player.js, it looks like Player object must have out of date position information in order for bubble to render in the wrong place. Don't understand the render code enough to know why the bubble would appear in this case, but not the corresponding plane which is placed according to the same data fields

**Comment by wight-airmash** [comment]
When player A hides behind the field of vision of player B, the high resolution position of A stops updating. But these coordinates are used to draw "say bubble":
https://github.com/airmash-refugees/airmash-frontend/blob/4ecd597ba3cba231b177a0656b7aa9ac9242e48a/src/js/Player.js#L697

Therefore, there are situations when it is drawn by the last known coordinates rather than by the correct ones. @derpsderps69 was right about that, but the problem will not dissapear completely with the max scale factor limit. This is only the more noticeable the smaller the server scale factor limit.

At the same time, I don't remember having this on the original server. This suggests that the current mass broadcasting of SAY packets is not quite correct.

I see two possible solutions:
1. Server side fix. Simple: send SAY packets to the players in the field of author vision only. Right way: the server should remember that in the last 4 seconds the player sent the SAY packet and personally duplicate this packet for all players who within these 4 seconds will appear in the field of author view.
2. Client side fix. Add drawing conditions to the frontend code, like this:

```js
if (this.render) {
  if (this.state.bubble) {
    this.sprites.bubble.visible = true;
  }

  this.sprites.bubble.position.set(this.pos.x * game.scale + e, (this.pos.y - n) * game.scale + t)
} else {
  this.sprites.bubble.visible = false;
}
```

**Comment by congratulatio** [comment]
Great analysis! Per your suggestion I've fixed this on the client side in https://github.com/airmash-refugees/airmash-frontend/commit/87dd3e0b176b4046f72b860903018de58a612e97, and will deploy it on https://airmash.online soon.

It's still an issue on https://starma.sh, but I've applied the same fix to my test fork of StarMash (which also has the sign-in feature and some other bug fixes) for Bombita to look over when he's got some time after his vacations.
